internal/service/appstream/stack_user_association.go

```diff
@@ -0,0 +1,152 @@
+package appstream
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appstream"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceUserStackAssociation() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceUserStackAssociationCreate,
+		ReadWithoutTimeout:   resourceUserStackAssociationRead,
+		DeleteWithoutTimeout: resourceUserStackAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"authentication_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(appstream.AuthenticationType_Values(), false),
+			},
+			"stack_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"user_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceUserStackAssociationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).AppStreamConn
+
+	input := &appstream.UserStackAssociation{
+		AuthenticationType: aws.String(d.Get("authentication_type").(string)),
+		StackName:          aws.String(d.Get("stack_name").(string)),
+		UserName:           aws.String(d.Get("user_name").(string)),
+	}
+
+	var output *appstream.BatchAssociateUserStackOutput
```

Comment:
Declaring the variable `output` here isn't needed, since the `:=` on the next line will declare it